### PR TITLE
Allow editor to access automations [MAILPOET-5024]

### PR DESCRIPTION
**Pull request by websupporter**
## Description
This PR enables editors to use the automation feature.

## QA
When you test this PR, deactivate and activate the plugin, so the capabilities of the editor get updated. Test with the editor role.

## Linked tickets
[MAILPOET-5024]

[MAILPOET-5024]: https://mailpoet.atlassian.net/browse/MAILPOET-5024?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

## After merge
Deactivate and activate plugin, when you want to reap the benefits immediately.

### CODE REVIEW

File: mailpoet/tests/integration/REST/Automation/Automations/AutomationPutEndpointTest.php [hunk 1]
```diff
@@ -31,6 +31,29 @@ public function _before() {
...
+          'name' => 'Test',
+        ],
+      ]
+    );
+
+    $this->assertNotSame([
+      'code' => 'rest_forbidden',
+      'message' => 'Sorry, you are not allowed to do that.',
+      'data' => ['status' => 401],
+    ], $data);
```
Comment (JanJakes) [inline mailpoet/tests/integration/REST/Automation/Automations/AutomationPutEndpointTest.php]: This check doesn't seem to make much sense given the check just below (and in all of the other occurrences) but I guess no harm in keeping it either.

File: mailpoet/lib/Config/AccessControl.php [hunk 2]
```diff
@@ -13,7 +12,7 @@ class AccessControl {
   const PERMISSION_MANAGE_SUBSCRIBERS = 'mailpoet_manage_subscribers';
   const PERMISSION_MANAGE_FORMS = 'mailpoet_manage_forms';
   const PERMISSION_MANAGE_SEGMENTS = 'mailpoet_manage_segments';
-  const PERMISSION_MANAGE_AUTOMATIONS = Engine::CAPABILITY_MANAGE_AUTOMATIONS;
+  const PERMISSION_MANAGE_AUTOMATIONS = self::PERMISSION_MANAGE_EMAILS;
```
Comment (JanJakes) [inline mailpoet/lib/Config/AccessControl.php]: We seem to be quite granular in the roles in general (managing subscribers, forms, segments, ...). Seeing that, isn't removing the automations role a bit of an inconsistency? Wouldn't it be better to just add the `editor` role as a default (same as for emails and for plugin admin)?

Comment (websupporter) [reply mailpoet/lib/Config/AccessControl.php]: > Wouldn't it be better to just add the editor role as a default (same as for emails and for plugin admin)?

~~Tbh, I tried that. For some reason, it did not work as expected~~

Ahhh! Capabilties are updated during plugin update and not on runtime! Yes, I agree, that would be better. Preparing...

**Comment by websupporter** [comment]
Thank you @JanJakes! Reset and recommitted.

**Review by JanJakes — APPROVED** [review]

**Review by Aschepikov — APPROVED** [review]
GJ!

**Pull request merged**